File: python/b_32373.py
import sys
sys.stdin = open('input.txt')

n,k = map(int,input().split())

arr = list(map(int,input().split()))

first = sorted(arr)

if arr == first:
    print('Yes')

else:
    for i in range(n-k):
        if arr[i] != first[i]:
            (arr[i], arr[i+k]) = (arr[i+k], arr[i])

    if arr == first:
        print('Yes')
    else:
        print('No')

File: C++/Cpp_Baekjoon/Cpp_Baekjoon/b_32373.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    
    int n, k;
    
    cin >> n >> k;

    vector<int> arr(n);
    
    for (int i = 0; i < n; i++) cin >> arr[i];

    for (int i = 0; i < arr.size(); i++) {
        if (i % k != arr[i] % k) {
            cout << "No";
            return 0;
        }
    }

    cout << "Yes";
   
    return 0;
}

